vpr/src/route/rr_graph2.cpp

```diff
@@ -982,15 +990,15 @@ static void load_block_rr_indices(const DeviceGrid& grid,
 
                 //Assign indices for SINKs and SOURCEs
                 // Note that SINKS/SOURCES have no side, so we always use side 0
-                for (const auto& class_inf : type->class_inf) {
-                    auto class_type = class_inf.type;
+                for (size_t iclass = 0; iclass < type->class_inf.size(); ++iclass) {
+                    auto class_type = type->class_inf[iclass].type;
                     if (class_type == DRIVER) {
-                        indices[SOURCE][x][y][0].push_back(*index);
-                        indices[SINK][x][y][0].push_back(OPEN);
+                        rr_graph_builder.node_lookup().add_node(RRNodeId(*index), x, y, SOURCE, iclass, SIDES[0]);
```

Comment:
Actually, I prefer to create the ``RRGraphBuilder::add_node()`` API in later pull request.
The detailed reasons are:
- The ``RRGraphBuilder::add_node()`` API is not a simple chained call using ``RRSpatialLookup``. It will have two functionality: (a) add a node to the ``node_storage``; and (b) register the node in the ``RRSpatialLookup``.
- Current builder functions are written in the way that the addition to ``node_storage`` is separated from the addition to  ``RRSpatialLookup``. If we created the ``RRGraphBuilder::add_node()`` API, it will cause massive changes on the  builder functions. So, I created the new API, it will not be useful in builder functions without changing.

That is why I prefer to do it in later pull request.

Reply:
SGTM.